test/unit/logger-mixin.test.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {LoggerMixin} from '../..';
+import {Application, Component} from '@loopback/core';
+import {Constructor} from '@loopback/context';
+
+describe('LoggerMixin', () => {
+  it('mixed class has .logger()', () => {
+    const myApp = new AppWithLogger();
+    expect(typeof myApp.logger).to.be.eql('function');
+  });
+
+  it('binds Logger from constructor', () => {
+    const myApp = new AppWithLogger({
+      loggers: [MyLogger],
+    });
+
+    expectLoggerToBeBound(myApp);
+  });
+
+  it('binds Logger from app.logger()', () => {
+    const myApp = new AppWithLogger();
+
+    expectLoggerToNotBeBound(myApp);
+    myApp.logger(MyLogger);
+    expectLoggerToBeBound(myApp);
+  });
+
+  it('binds user defined component without Logger', () => {
+    class EmptyTestComponent {}
+
+    const myApp = new AppWithLogger({
+      components: [EmptyTestComponent],
+    });
+
+    expectComponentToBeBound(myApp, EmptyTestComponent);
+  });
```

Comment:
I find this test extraneous - the fact that Application's constructor invokes `app.component()` is already tested in `@loopback/core`. No need to test it again here.